### PR TITLE
Fixes Issue 269

### PR DESCRIPTION
**Pull request by jorgejesus**
Based on the discussion on #269  Have implemented the following configuration on the postgresql driver:

```
        provider:
            name: PostgreSQL
            data:
                host: 127.0.0.1
                dbname: test
                user: postgres
                password: postgres
                port: 5432
                search_path: [osm,public]
            id_field: osm_id
            table: hotosm_bdi_waterways
            geom_field: foo_geom
```

There is no `schema` parameters since using the proposed search path `[x, public] ` will ensure that tables are found and functions in PostGIS are properly called

The dataset  `hotosm_bdi_waterways` has been updated to use a specific schema and defined geometry column. 

If geometry column is not defined it will be set to 'geom' and if no search path is set then it will be public

### CODE REVIEW

**Comment by jorgejesus** [comment]
Also made a local test with the following configuration
```
        provider:
            name: PostgreSQL
            data:
                host: 127.0.0.1
                dbname: test
                user: postgres
                password: postgres
                port: 5432
            id_field: osm_id
            table: hotosm_bdi_waterways

```

File: pygeoapi/provider/postgresql.py [hunk 1]
```diff
@@ -87,18 +90,18 @@ def __init__(self, conn_dic, table, context="query"):
...
         self.columns = None
         self.conn = None
-        self.schema = None
 
     def __enter__(self):
         try:
-            self.schema = self.conn_dic.pop('schema', None)
-            if self.schema == 'public' or self.schema is None:
+            self.search_path = self.conn_dic.pop('search_path', None)
+            if self.search_path is ['public'] or self.search_path is None:
```
Comment (alpha-beta-soup) [inline pygeoapi/provider/postgresql.py]:
```
>>> ['public'] is ['public']
False
```

File: pygeoapi/provider/postgresql.py [hunk 2]
```diff
@@ -87,18 +90,18 @@ def __init__(self, conn_dic, table, context="query"):
...
                 pass
             else:
-                self.conn_dic["options"] = '-c search_path={}'.format(
-                    self.schema)
-                LOGGER.debug('Using schema {} as search path'.format(
-                    self.schema))
+                if "public" not in self.search_path:
+                    self.search_path = self.search_path + ["publc"]
```
Comment (alpha-beta-soup) [inline pygeoapi/provider/postgresql.py]: Typo: `publc`; and do we always want to add this schema? Why?

File: pygeoapi/provider/postgresql.py [hunk 3]
```diff
@@ -149,6 +152,8 @@ def __init__(self, provider_def):
         self.table = provider_def['table']
         self.id_field = provider_def['id_field']
         self.conn_dic = provider_def['data']
+        self.geom = provider_def.get('geom_field') \
+            if provider_def.get('geom_field') else "geom"
```
Comment (alpha-beta-soup) [inline pygeoapi/provider/postgresql.py]: Why not `self.geom = provider_def.get('geom_field', 'geom')` ?

File: pygeoapi/provider/postgresql.py (hunk 2 above)
Comment (alpha-beta-soup) [reply pygeoapi/provider/postgresql.py]: I think `['public']` should be the default if the property is completely unspecified, but I don't think it needs to be enforced if it's explicitly left out from the configuration.

File: .gitignore [hunk 4]
```diff
@@ -111,3 +111,7 @@ openapi.yml
 
 #Eclipse project
 .project
+docker-compose.yml
```
Comment (tomkralidis) [inline .gitignore]: Are the `.gitignore` updates intentional? Files listed seem random-ish>

Comment (jorgejesus) [reply .gitignore]: Mijn fout it was not supposed to be pushed

File: pygeoapi/provider/postgresql.py (hunk 1 above)
Comment (jorgejesus) [reply pygeoapi/provider/postgresql.py]: @alpha-beta-soup  Thank you, now I know why this is wrong (http://net-informations.com/python/iq/is.htm)

File: pygeoapi/provider/postgresql.py (hunk 2 above)
Comment (justb4) [reply pygeoapi/provider/postgresql.py]: Agree (do not add `public` if missing in `search_path`) .

File: pygeoapi/provider/postgresql.py (hunk 1 above)
Comment (justb4) [reply pygeoapi/provider/postgresql.py]: Yes `is` and `==` can be tricky in Python 😈. Also I tend to never have `else` clauses, many refs on web like recently on [Medium](https://medium.com/better-programming/why-you-need-to-stop-using-else-statements-5b1fd09dea9e). Just a suggestion, halves number of lines, but consider this as syntactic sugar: 
```
search_path = self.conn_dic.pop('search_path', ['public'])
if search_path != ['public']:
    self.conn_dic["options"] = f'-c search_path={",".join(search_path)}'
    LOGGER.debug(f'Using explicit search path: {search_path}')
self.conn = psycopg2.connect(**self.conn_dic)
```
(IMO no need for `search_path` to be member var, in that case would also need initialization in `__init__()`))

Comment (jorgejesus) [reply pygeoapi/provider/postgresql.py]: @justb4  Thank you for the fresh eyes, I have implemented your suggestions

File: pygeoapi/provider/postgresql.py (hunk 3 above)
Comment (jorgejesus) [reply pygeoapi/provider/postgresql.py]: @alpha-beta-soup  Implemented recommendation, thank you

File: .gitignore (hunk 4 above)
Comment (jorgejesus) [reply .gitignore]: `.gitignore`has is now identical to master

**Comment by jorgejesus** [comment]
@justb4  Well spotted !!! There is already the log information inside `__enter__` so better to remove the extra logging

**Pull request merged**